File: WebNotePHP_Start/module/Application/src/Application/Controller/NoteController.php
<?php

namespace Application\Controller;

use Application\Repository;
use Zend\Mvc\Controller\AbstractRestfulController;
use Zend\View\Model\JsonModel;

class NoteController extends AbstractRestfulController {

    private $repository;

    public function __construct() {
        $this->repository = new NoteRepository();
    }

    // GET
    public function getList() {
        return new JsonModel($this->repository->readAll());
    }

    // GET
    public function get($id) {
        return new JsonModel($this->repository->read($id));
    }

    // POST
    public function create($data) {

        $emptyNote = array(
            'Title' => "",
            'Message' => "",
            'Added' => new \MongoDate(),
            'Categories' => []
        );

        $newId = $this->repository->create($emptyNote);
        return new JsonModel([$newId]);
    }

    // PUT
    public function update($id, $data) {
        $this->repository->update($data);
        return new JsonModel();
    }

    // DELETE
    public function delete($id) {
        $this->repository->delete($id);
        return new JsonModel();
    }
}

File: WebNote/WebNote/Controllers/NoteController.cs
using System.Collections.Generic;
using System.Web.Http;
using DeveloperMediaDemo.Models;

namespace DeveloperMediaDemo.Controllers
{
    public class NoteController : ApiController
    {
        private NoteRepository repository;

        public NoteController()
        {
            repository = new NoteRepository();
        }

        public IEnumerable<Note> GetAll()
        {
            return repository.ReadAll();
        }

        public Note Get(string id)
        {
            return repository.Read(id);
        }

        public string Post()
        {
            var newNote = new Note();
            repository.Create(newNote);
            return newNote.Id;
        }

        public void Put(Note note)
        {
            repository.Update(note);
        }

        public void Delete(string id)
        {
            repository.Delete(id);
        }
    }
}